### PR TITLE
put some reasonable limits on SPORK_12_RECONSIDER_BLOCKS

### PR DESCRIPTION
**Pull request by UdjinM6**
Humans are prone to errors and this feature is way too powerful to be executed "AS IS". Put some limits on it: we don't want to reprocess way too many blocks and we don't want to be able to do that way too often.

### CODE REVIEW

**Comment by tgflynn** [comment]
6 hours seems a bit of a short time to me to be sure a problem is known by one of the 3 people who have the spork key.  I would suggest 24 hours.

**Comment by UdjinM6** [comment]
Agree

File: src/spork.cpp [hunk 1]
```diff
@@ -75,9 +75,28 @@ void CSporkManager::ExecuteSpork(int nSporkID, int nValue)
...
     //correct fork via spork technology
     if(nSporkID == SPORK_12_RECONSIDER_BLOCKS && nValue > 0) {
+        // allow to reprocess 24h of blocks max, which should be enough to resolve any issues
+        int64_t nMaxBlocks = 576;
+        // this potentially can be a heavy operation, so only allow this to be executed once per 10 minutes
+        int64_t nTimeout = 10 * 60;
+
+        static int64_t nTimeExecuted = 0; // i.e. it was never executed before
+
+        if(GetTime() - nTimeExecuted < nTimeout) {
```
Comment (tgflynn) [inline src/spork.cpp]: It looks like this will never allow execution because GetTime() >> 0.

Comment (tgflynn) [inline src/spork.cpp]: N/M I was wrong.

**Pull request merged**